### PR TITLE
feature add search view favorite

### PR DESCRIPTION
**Pull request by HyperionsThs**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -23,8 +28,9 @@
         </activity>
         <activity
             android:name=".ui.heroinfo.HeroInfoActivity"
+            android:configChanges="orientation"
```
Comment (daolq3012) [inline app/src/main/AndroidManifest.xml]: why u add this line? pls declare

File: app/src/main/java/com/hyperion/ths/marvel_03/ui/favorite/FavoriteViewModel.java [hunk 2]
```diff
@@ -38,6 +40,8 @@ public FavoriteViewModel(FavoriteFragmentAdapter favoriteFragmentAdapter,
         mHeroRepository = heroRepository;
         mNavigator = navigator;
         mDialogManager = dialogManager;
+        MainActivity mainActivity = (MainActivity) navigator.getActivity();
+        mainActivity.setOnTextSearchFavoriteListener(this);
```
Comment (daolq3012) [inline app/src/main/java/com/hyperion/ths/marvel_03/ui/favorite/FavoriteViewModel.java]: change to
```java
 Activity activity = navigator.getActivity();
if(activity instanceof MainActivity){
 ((MainActivity)activity).setOnTextSearchFavoriteListener(this);
}
```

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (HyperionsThs) [inline app/src/main/AndroidManifest.xml]: With android:configChanges="orientation" you tell Android that you will be responsible of the changes of orientation.

Comment (daolq3012) [inline app/src/main/AndroidManifest.xml]: pls remove if u add ``android:screenOrientation="portrait"`` 
this line uneccessary

File: app/src/main/java/com/hyperion/ths/marvel_03/ui/favorite/FavoriteFragmentAdapter.java [hunk 3]
```diff
@@ -65,6 +73,36 @@ public int getItemCount() {
...
+                String charString = charSequence.toString();
+                if (charString.isEmpty()) {
+                    mHeroList = mHeroListStore;
+                } else {
+                    List<Hero> filteredList = new ArrayList<>();
+                    for (Hero hero : mHeroListStore) {
+                        if (hero.getName().toLowerCase(Locale.getDefault()).contains(charString)) {
+                            filteredList.add(hero);
+                        }
+                    }
```
Comment (daolq3012) [inline app/src/main/java/com/hyperion/ths/marvel_03/ui/favorite/FavoriteFragmentAdapter.java]: change to
```java
else { 
                    mHeroList.clear();
                    for (Hero hero : mHeroListStore) {
                        if (hero.getName().toLowerCase(Locale.getDefault()).contains(charString)) {
                            mHeroList.add(hero);
                        } 
                    } 
                } 
```

**Pull request merged**